### ISSUE TITLE
BulkInsertOrUpdate: There is already an object named 'DogTempab3020de' in the database.

### ISSUE THREAD

**Issue by tpdickson** [borisdj/EFCore.BulkExtensions#142]
I'm getting an issue with `BulkInsertOrUpdate` with one of my tables. I have two tables, Cat and Dog (not actual names). The models for Cat and Dog are fairly similar (column names replaced):

- Cat model: https://gist.github.com/tpdickson/6992ffc7d56f36b1e5ee26dc5752882d

- Dog model: https://gist.github.com/tpdickson/9ec3347adee03aca1a932aee097bad12

Both are set up in very similar ways in the DB context:

https://gist.github.com/tpdickson/4b70eb3e8df80c17eb532c06fdd612cb

The data is being read in from .csv files as `IEnumerable<>` types, then converted `ToList()`, then passed into `context.BulkInsertOrUpdate()`:

```csharp
var dataFromFile = ReadCsvFile().ToList();
var db = new CatDogContext();
db.BulkInsertOrUpdate(dataFromFile);
```

**When I read and insert Cat data, everything completes without issue.**

**When I read and insert Dog data, it fails as soon as `db.BulkInsertOrUpdate()` is called, with the following output:**

`There is already an object named 'DogTempab3020de' in the database.`

https://gist.github.com/tpdickson/5b648d48d2c9a3a5b18ac6bf885e5ffd

I've tried using `BulkConfig.UseTempDB = false` and various `BulkConfig.BatchSize` values without any luck. The only obvious difference I can see between Cat and Dog is that Dog has some `double` or `float` fields, while Cat doesn't. Besides that they're seemingly the same.

Any guesses on what might be causing the issue? Thanks.

Microsoft.NETCore.App 2.2.0
EFCore.BulkExtensions 2.3.7
Microsoft.EntityFrameworkCore.SqlServer 2.2.0
Microsoft.EntityFrameworkCore.Tools 2.2.0

**Comment by tpdickson** [comment]
The issue seems to stem from the `class CatDogContext : DbContext` implementation. The line that's causing it is `.HasColumnName("Column10[]]")`. In the SQL script for the table, that column is created as `[Column10[]]] [nvarchar](255) NULL` which works properly to create column name `Column10[]`.

If I use `.HasColumnName("Column10[]]")` EF Core seems to read in the column name properly, yet for some reason fumbles and outputs the error from above when it seemingly tries to create multiple copies of the same temp table.

If I use `.HasColumnName("Column10[]")` EF Core concatenates `Column10[]` with `, T.[Column11` and errors out (as expected).

If I use `.HasColumnName("Column10[]]]")` EF Core also concatenates `Column10[]` with `, T.[Column11` and errors out.

So I think this actually might be a bug with EF Core and how it parses column names that contain `[` or `]`. Either that or I'm doing something wrong.

**Comment by borisdj** [comment]
Try NuGet version 2.3.9
Should now  be fixed.

**Comment by tpdickson** [comment]
Seems to be fixed in 2.3.9 by using `.HasColumnName("Column10[]")`.

Thanks a ton.
